core/src/test/java/org/apache/druid/testing/DeadlockDetectingFailOnTimeout.java

```diff
@@ -44,6 +44,7 @@
  * This class is based on {@link org.junit.internal.runners.statements.FailOnTimeout}, additionally deadlocked
  * threads are detected.
  */
+@SuppressWarnings("SSBasedInspection") // Prohibit check on Thread.getState()
```

Comment:
Please annotate just the `private Thread getStuckThread(Thread mainThread)` method rather than the whole class.